### ISSUE TITLE
Master.DSK

### ISSUE THREAD

**Issue by TommyGH** [AppleWin/AppleWin#438]
Master.DSK

What exactly is this disk image?
From where did it originate?
Is it documented in AppleWin help?
Is it documented in GITHUB?

The name "Master" might lead me to believe it is one of the following:

  DOS 3.3 System Master     (From 1980)
  DOS 3.3 System Master     (From 1983)

Master.DSK is not one of those 2 disk images.  In fact, there are code differences between Master.DSK and those 2 images.  At times this has caused me some confusion.

For example, on T0 S4 is the beginning of the code for MSWAIT 

DOS 3.3 System Master has this:

  BA00-   A2 11       LDX   #$11        
  BA02-   CA            DEX   
  BA03-   D0 FD       BNE   $BA02
  BA05-   E6 4E        INC   $4E
  BA07-   D0 02       BNE   $BA0B
  BA09-   E6 47        INC   $47
  BA0B-   38             SEC   
  BA0C-   E9 01       SBC   #$01
  BA0E-   D0 F0       BNE   $BA00
  BA10-   60            RTS   

While MASTER.DSK has:

  BA00-   EA          NOP   
  BA01-   EA          NOP   
  BA02-   EA          NOP   
  BA03-   EA          NOP   
  BA04-   EA          NOP   
  BA05-   E6 4E      INC   $4E
  BA07-   D0 02     BNE   $BA0B
  BA09-   E6 47     INC   $47
  BA0B-   38          SEC   
  BA0C-   E9 01     SBC   #$01
  BA0E-   D0 F0     BNE   $BA00
  BA10-   60          RTS   

One place to see DOS3.3 documentation is:

https://htmlpreview.github.io/?https://github.com/Michaelangel007/apple2_dos33/blob/master/dos33.html

I seem to recall a long time ago a discussion that indicated there was need for a "special" DOS image to make AppleWin more efficient.  I'm not sure if I am remembering that correctly or if it even applies here.

What do I expect by opening this issue?  Possibilities:

1. nothing
2. explanation about Master.DSK and why it is different from a normal DOS image.
3. documentation in GITHUB (history, readme, help, etc.)
4. a change to Master.DSK in GITHUB so that it matches one of the DOS3.3 images.
5. other

Tommy

**Comment by sicklittlemonkey** [comment]
Good find Tommy. We should replace or remove this. Tom might know, or
Oliver.

The disk controller ROM also used to have the delay code NOPed out, until I
had a problem with The CIA Files which checksums the ROM.
​
Cheers,
Nick.

**Comment by tomcw** [comment]
This pre-dates both me and @oliverschmidt.
I went back to O'Brien's original beta2 release from March 1995, and that MASTER.DSK is binary identical to the one we ship today (1.26.2.4).

**Comment by oliverschmidt** [comment]
I didn't even know that it was "special" until now. I would never have added any image because of IP issues but as it was "always" there I didn't dare to remove it as I thought that this might upset the users.

**Comment by Michaelangel007** [comment]
Removing it is NOT an option.  The _whole_ point of AppleWin is that things just work "Out-of-the-Box" without forcing the user to waste _their_ time downloading some obscure 140 KB binary file. (i.e. /me glares at Virtual ][)

We _should_ be using a "vanilla" version -- If that doesn't run then then we need to fix the emulator.

The official DOS 3.3 source code in question is:

```
3A00:A2 11          031 MSWAIT      LDX #$11           
3A02:CA             032 MSW1        DEX             DELAY 86 USEC.
3A03:D0 FD          033             BNE MSW1
3A05:E6 46          034             INC MONTIMEL
3A07:D0 02          035             BNE MSW2        ; DOUBLE-BYTE
3A09:E6 47          036             INC MONTIMEH    INCREMENT.
3A0B:38             037 MSW2        SEC
3A0C:E9 01          038             SBC #$1         DONE 'N' INTERVALS?
3A0E:D0 F0          039             BNE MSWAIT            (A-REG COUNTS)
3A10:60             040             RTS
                    041 AEC1        EQU *           ;TELL RELOCATOR WHERE CORE ENDS
```

**Comment by Michaelangel007** [comment]
Tommy,

Just a gentle reminder to NOT share my DOS 3.3 source code listing for the time being until I have all the internal variables and functions hyper-linked, aka, marked up.

Thanks.

**Comment by TommyGH** [comment]
I agree that there should be at least one ".DSK" image supplied with AppleWin.

By the way, MASTER.DSK is badly named as it is actually a SLAVE diskette initially loaded to $9D00 during the boot process.

And almost any change to MASTER.DSK for AppleWin will cause someone in the future to be confused.  For example, if the name is left the same, but the contents of the image are different, someone somewhere will have both versions on their PC and wonder why they are different.  It's kind of like the problem of calling your official version of your operating system "DOS3.3", but then having to qualify it with "the one from 1980" or "the one from 1983".

Are there any issues with packaging the official Apple DOS3.3(1983) master disk image with AppleWin?  ...and clearly identifying it as such in the name of the DSK image?

**Comment by TommyGH** [comment]
MASTER.DSK has a lot of differences from a normal DOS3.3 INIT'd diskette besides MSWAIT code difference I mentioned earlier.

For example, if you boot MASTER.DSK, and then type "INT" for switching to INTEGER BASIC, you get the message "ERROR #1" instead of "LANGUAGE NOT AVAILABLE".  It is interesting that #1 is the error number for that message (as documented in DOS3.3 and DOS3.2 manuals).

Hmmmm.  I just did some more investigation.  If I boot "Diversi-copy v3.2.dsk" (From Asimov), select option 5 to "Exit to Basic", and then enter "INT" I get the same "ERROR #1" message.  As I noted somewhere else, the "Conflict in Vietnam" diskette uses a version of DiversiDos and has a similar boot load as MASTER.DSK.

**Comment by tomcw** [comment]
>Are there any issues with packaging the official Apple DOS3.3(1983) master disk image with AppleWin? ...and clearly identifying it as such in the name of the DSK image?

So how about replacing with this version:
https://mirrors.apple2.org.za/ftp.apple.asimov.net/images/masters/Apple%20DOS%203.3%20January%201983.dsk

![dos3 3 master jan-1-1983](https://user-images.githubusercontent.com/6696896/29288244-f5dd1b36-812f-11e7-9d3c-43a1c3ad2ceb.png)

**Comment by TommyGH** [comment]
I've recently discovered that there are 3 versions of DOS3.3:

  1980
  1983 January
  1983 August

The link in the previous post (by the other Tom), which has a boot screen saying "January" is the one from later in 1983 (August) .  (I know this because it has the "patch" that was documented as July 11, 1983).  I think the official Apple released version from "August" displayed "August" on the boot screen.

https://mirrors.apple2.org.za/ftp.apple.asimov.net/images/masters/Apple%20DOS%203.3%20January%201983.dsk

Here is at least one reference to saying there were 3 versions released for DOS3.3:

https://github.com/cmosher01/Apple-II-System-Masters

Unfortunately the README for cmosher01 APPLEII-SYSTEM-MASTERS says he got his DOS3.3 August image from Asimov, but the image DOES display "August, 1983" when booted.  It also DOES have the "FIXIT2" code at $B6B3 (for those in the know).

**Comment by TommyGH** [comment]
https://github.com/cmosher01/Apple-II-Source/blob/master/README-notes

From Apple Assembly Line, Vol. 3, Issue 12, Sept. 1983:
 
110 
 
111 Yet Another New Version of DOS 3.3 	Bob Sander-Cederlof
 
112 
 
113 In the July issue of AAL I outlined the changes Apple made to DOS 3.3 early this year.
 
114 Today I received a new "Developer's System Master", with a cover letter claiming another
 
115 correction to the APPEND routine. The letter binds developers to begin using the new
 
116 version no later than November 1st.
 
117 
 
118 If you like APPEND, or would like to like it, you might want to make these patches in
 
119 your own system master. I am going to assume you already have the "early 1983" version,
 
120 either because you bought a //e or a disk drive this year, or you copied one from a friend,
 
121 or you made the patches from my July article. Here are the new changes:
 
(snip)

 
140 What Apple has done is move the patch they had put at $BA84 down to $B6B3 and added four
 
141 extra lines to that patch. I HOPE IT IS NOW CORRECT!

**Comment by Michaelangel007** [comment]
And back in Real World zero fucks are given about an 39-year Operating System that only hardcore fanatics still use today.

**Comment by sicklittlemonkey** [comment]
I'm bumping this issue because of a recent discussion where @Michaelangel007 again talked about DOS versions.

And mentioning @TommyGH in case Tom fixing #290 might inspire further Great Works, faster!
